### PR TITLE
Sign Up

### PR DESCRIPTION
**Pull request by abdullahallmehedi**

### CODE REVIEW

File: app/assets/javascripts/users.coffee [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
```
Comment (chuanhtuan) [inline app/assets/javascripts/users.coffee]: remove unused file

File: app/assets/stylesheets/users.scss [hunk 2]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the users controller here.
```
Comment (chuanhtuan) [inline app/assets/stylesheets/users.scss]: remove unused file

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -0,0 +1,49 @@
+class UsersController < ApplicationController
+  def index
+    @users = User.paginate(page: params[:page])
```
Comment (chuanhtuan) [inline app/controllers/users_controller.rb]: remove ()

File: app/controllers/users_controller.rb [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  end
+  
+  def show     
+    if params.has_key?(:id)
```
Comment (chuanhtuan) [inline app/controllers/users_controller.rb]: no need check id

File: app/controllers/users_controller.rb [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+      @user = User.find(params[:id])
```
Comment (chuanhtuan) [inline app/controllers/users_controller.rb]: remove ()

File: app/controllers/users_controller.rb [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+      redirect_to new_user_path
+    end
+  end
+  
+  def new
+    @user = User.new
+  end
+  
+  def create
+    @user = User.new(user_params)
```
Comment (chuanhtuan) [inline app/controllers/users_controller.rb]: remove ()

File: app/controllers/users_controller.rb [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+    @user.active = true
+    @user.rank = User.count + 1
+    if @user.save
+      flash[:success] = "Thank you for joining with us."
```
Comment (chuanhtuan) [inline app/controllers/users_controller.rb]: i18n

File: app/controllers/users_controller.rb [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+  end
+  
+  def new
+    @user = User.new
+  end
+  
+  def create
+    @user = User.new(user_params)
+    @user.active = true
+    @user.rank = User.count + 1
```
Comment (chuanhtuan) [inline app/controllers/users_controller.rb]: what is "rank"?

File: app/controllers/users_controller.rb [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+    if @user.save
+      flash[:success] = "Thank you for joining with us."
+      redirect_to @user
+    else
+      render :new
+    end
+  end
+  
+  def edit
+    @user = User.find(params[:id])
```
Comment (chuanhtuan) [inline app/controllers/users_controller.rb]: ()

File: app/controllers/users_controller.rb [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+  end
+  
+  def update
+    @user = User.find(params[:id])
```
Comment (chuanhtuan) [inline app/controllers/users_controller.rb]: ()

File: app/controllers/users_controller.rb [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+    if @user.update_attributes(user_params)
```
Comment (chuanhtuan) [inline app/controllers/users_controller.rb]: ()

File: app/controllers/users_controller.rb [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+      flash[:success] = "Profile updated"
```
Comment (chuanhtuan) [inline app/controllers/users_controller.rb]: i18n

File: app/views/users/edit.html.erb [hunk 13]
```diff
@@ -0,0 +1,26 @@
+<h1>Update your profile</h1>
+
+<div class="row">
+  <div class="col-xs-6 col-xs-offset-3">
+    <%= form_for(@user) do |f| %>
```
Comment (chuanhtuan) [inline app/views/users/edit.html.erb]: ()

File: app/views/users/edit.html.erb [hunk 14]
```diff
@@ -0,0 +1,26 @@
...
+      <%= f.label :email %>
+      <%= f.email_field :email, class: 'form-control' %>
+      
+      <%= f.label :password %>
+      <%= f.password_field :password, class: 'form-control' %>
+      
+      <%= f.label :password_confirmation, "Confirmation" %>
+      <%= f.password_field :password_confirmation, class: 'form-control' %>
+      
+      <%= f.submit "Save changes", class: "btn btn-primary" %>
```
Comment (chuanhtuan) [inline app/views/users/edit.html.erb]: i18n

File: app/views/users/new.html.erb [hunk 15]
```diff
@@ -0,0 +1,26 @@
+<h1>Sign up</h1>
+
+<div class="row">
+  <div class="col-xs-6 col-xs-offset-3">
+    <%= form_for(@user) do |f| %>
```
Comment (chuanhtuan) [inline app/views/users/new.html.erb]: ()

File: app/views/users/new.html.erb [hunk 16]
```diff
@@ -0,0 +1,26 @@
...
+      <%= f.label :email %>
+      <%= f.email_field :email, class: 'form-control' %>
+      
+      <%= f.label :password %>
+      <%= f.password_field :password, class: 'form-control' %>
+      
+      <%= f.label :password_confirmation, "Confirmation" %>
+      <%= f.password_field :password_confirmation, class: 'form-control' %>
+      
+      <%= f.submit "Register", class: "btn btn-primary" %>
```
Comment (chuanhtuan) [inline app/views/users/new.html.erb]: i18n

File: app/controllers/users_controller.rb (hunk 5 above)
Comment (abdullahallmehedi) [inline app/controllers/users_controller.rb]: What to remove here?

File: app/controllers/users_controller.rb (hunk 6 above)
Comment (abdullahallmehedi) [inline app/controllers/users_controller.rb]: What to remove here?

File: app/controllers/users_controller.rb (hunk 8 above)
Comment (abdullahallmehedi) [inline app/controllers/users_controller.rb]: As per your requirement:
[Framgia E-Learning System]
【Home】
User
- can see his summary (follow, the number of words his memorised, ranking, etc.)

File: app/controllers/users_controller.rb (hunk 5 above)
Comment (chuanhtuan) [inline app/controllers/users_controller.rb]: ()
about coding convention, do not use () here

File: app/controllers/users_controller.rb (hunk 8 above)
Comment (chuanhtuan) [inline app/controllers/users_controller.rb]: remove ranking :dancer: no need implement ranking

**Pull request merged**